lms/djangoapps/certificates/queue.py

```diff
@@ -291,8 +291,8 @@ def add_cert(self, student, course_id, course=None, forced_grade=None, template_
         mode_is_verified = enrollment_mode in GeneratedCertificate.VERIFIED_CERTS_MODES
         user_is_verified = IDVerificationService.user_is_verified(student)
         cert_mode = enrollment_mode
-        is_eligible_for_certificate = is_whitelisted or CourseMode.is_eligible_for_certificate(enrollment_mode,
-                                                                                               cert_status)
+        is_eligible_for_certificate = (is_whitelisted and enrollment_mode != CourseMode.AUDIT) or \
```

Comment:
According to course eligibility definition, a whitelisted learner is eligible for a certificate regardless of grade or enrollment track. I'm not sure if we are breaking any open edX scenario here, @nedbat could you please take a look?
https://edx.readthedocs.io/projects/edx-partner-course-staff/en/latest/student_progress/course_grades.html#interpret-the-grade-report